code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -1003,6 +1003,11 @@ GLOBAL_LIST_EMPTY(PDAs)
 		if(inserted_item)
 			to_chat(user, "<span class='warning'>There is already \a [inserted_item] in \the [src]!</span>")
 		else
+			if(istype(C, /obj/item/pen/screwdriver))
+				var/obj/item/pen/screwdriver/temp = C
+				if(temp.extended)
+					to_chat(user, "<span class='warning'>You can't fit the pen in!</span>")
+					return
```

Comment:
I dislike checking for specific types when doing general behaviours like this, stuff specific to an item should be contained within that item.
Perhaps some kind of var on the pen, or checking the size of the item and changing the item size when you extend the screwdriver would improve code quality